internal/internal_workflow_client.go

```diff
@@ -1450,7 +1450,7 @@ func (iter *historyEventIteratorImpl) HasNext() bool {
 func (iter *historyEventIteratorImpl) Next() (*historypb.HistoryEvent, error) {
 	// if caller call the Next() when iteration is over, just return nil, nil
 	if !iter.HasNext() {
-		panic("HistoryEventIterator Next() called without checking HasNext()")
```

Comment:
The code behaviour is correct the comment is incorrect, if you want you can submit a PR to update the comment

Reply:
Sg, updated to keep the panic and added a documentation comment to make the behavior and dependency on `HasNext` clear.

As a user of the API, my initial expectation was that reaching the end of the events would return an error like ErrNoMoreResults, rather than triggering a panic. I think for most go users' expectations, panics are generally an indicator for unrecoverable errors, not for a normal control flow situation like running out of events.

Reply:
>Sg, updated to keep the panic and added a documentation comment to make the behavior and dependency on HasNext clear.

Yes

Not calling `HasNext` before `Next ` is a logic error in your code like trying send to a closed channel, it is not control flow